### ISSUE TITLE
Changing network in google_container_cluster fails silently

### ISSUE THREAD

**Issue by nyurik** [hashicorp/terraform-provider-google#7390]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.13.3
+ provider registry.terraform.io/hashicorp/google v3.41.0

### Affected Resource(s)

* `google_container_cluster`

### Terraform Configuration Files

```tf
# First create a cluster:
resource "google_container_cluster" "cluster" {
  ...
}

# Afterwards, change it to this:
resource "google_compute_network" "network" {
  name = "foo"
}

resource "google_container_cluster" "cluster" {
  network = "foo"
  ...
}
```

### Expected Behavior

Network should have changed from `default` to `foo`

### Actual Behavior

Cluster's network stayed as `default`, but `terraform apply` did not report any errors.

**Comment by edwardmedia** [comment]
@nyurik I tested and couldn't repro it. What error did you receive? Can you post the debug log so I can take a close look?

**Comment by nyurik** [comment]
@edwardmedia I did not get any error, I only observed that after changing the network from default to a specific network selflink, the web UI still showed default.  Destroying and recreating the cluster fixed the issue. I won't be able to easily repo the issue though, plus obviously there were no logs.  One possible reason now that I think about it could be that web UI was not refreshing the status for a very long time (~10min), in which case this wouldn't be TF issue.
